core/src/integration-test/java/io/aiven/kafka/tieredstorage/RemoteStorageManagerTest.java

```diff
@@ -532,6 +533,7 @@ void testTransformingEmptyIndexes(final boolean encryption) {
             segmentIndexBuilder
         );
         assertThat(is).isEmpty();
+        assertThat(segmentIndexBuilder.indexes()).containsOnly(IndexType.OFFSET);
```

Comment:
Should we maybe add `segmentIndexBuilder.build()` here and check the result, at least that the indexed size is indeed 0?

Reply:
Sure, adding validations on the latest commit